File: src/main/java/com/hoi4utils/clausewitz/exceptions/UnexpectedLocalizationStatusException.java
package com.hoi4utils.clausewitz.exceptions;

import com.hoi4utils.clausewitz.localization.Localization;

public class UnexpectedLocalizationStatusException extends RuntimeException {
    public UnexpectedLocalizationStatusException(Localization localization, Localization other) {
        super("Unexpected localization statuses: " + localization + ", status: " + localization.status()
                + " and " + other + ", status: " + other.status());
    }

    public UnexpectedLocalizationStatusException(Localization localization) {
        super("Unexpected localization status: " + localization + ", status: " + localization.status());
    }
}

File: src/main/scala/com/hoi4utils/exceptions/UnexpectedLocalizationStatusException.scala
package com.hoi4utils.exceptions

import com.hoi4utils.hoi4.localization.Localization

class UnexpectedLocalizationStatusException(message: String) extends Exception(message):

  def this(loc: Localization, other: Localization) =
    this(
      s"Unexpected localization statuses: $loc, status: ${loc.status} and $other, status: ${other.status}"
    )

  def this(loc: Localization) =
    this(
      s"Unexpected localization status: $loc, status: ${loc.status}"
    )
